helper/helper.go

```diff
@@ -363,3 +366,27 @@ func TLScertToFile(filename string, derBytes []byte) error {
 
 	return err
 }
+
+// FormatUploadFilePath ensures that path separators are "/", and returns error if the
+// filepath contains a disallowed character matched with regex
+func FormatUploadFilePath(filePath string) (string, error) {
+
+	// Check for mixed "\" and "/" in filepath. Stop and throw an error if true so that
+	// we do not end up with unintended folder structure when applying ToSlash later on
+	if strings.Contains(filePath, "\\") && strings.Contains(filePath, "/") {
+		return filePath, fmt.Errorf("filepath contains mixed '\\' and '/' characters")
+	}
```

Comment:
```suggestion
	outPath := strings.ReplaceAll(filePath, "\\", "/")
```

If we can assume that file paths with double backslashes `\\` originate from a windows machine we could simplify it like this.

Reply:
I see the reasoning but I do not agree to change all backslashes automatically to slashes  since this can potentially cause a mess in the folder structure.
The above looks for paths of the form `path1\path2/path2\file.c4gh` etc. So it basically mitigates the recent problem we had with the `-targetDir` flag which was fixed in `sda-cli` but still it's worth to keep a check just in case this problem resurfaces in some other way. The suggested simplification is basically done at line 381 (see comment above).

Reply:
Can you clarify the use case for having backslashes in the object name?

Reply:
I wouldn't say it is a use case but rather a situation we wish to avoid happening again in the future.

Reply:
It seems to me the safest way to avoid the recent issues would be to remove all backslashes from the names -  I really don't see when one would want to not delete them.

Reply:
I agree, however I would prefer not to interfere with the filepath by deleting/renaming backslashes automatically. So, for now I would suggest we keep the `regex` rule where if a backslash is found the request is rejected.